test/example_test.dart

```diff
@@ -1,7 +1,29 @@
+import 'dart:io';
+
+import 'package:flutter_downloader/flutter_downloader.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
-  test('example test', () {
-    expect(2 + 2, 4);
-  });
+  test('example.com download', isolated((testDir) async {
+    const sampleUrl = 'https://www.example.com';
+    const sampleUrlHash = '740e7397907c0b004010d92b33d283e98f74063d';
+      await FlutterDownloader().startDownload(sampleUrl);
+      final metaFile = File('${testDir.path}/$sampleUrlHash.meta');
+      assert(metaFile.existsSync());
+  }),);
+}
+
+/// Run a test in a temp directory, which will be deleted after the execution.
+dynamic Function() isolated(Future<dynamic> Function(Directory testDir) body) {
```

Comment:
You might want to use [`package:file`](https://pub.dev/packages/file) for this - it abstracts the concrete file system and provides a useful `MemoryFileSystem` for use in testing.